server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -476,25 +476,50 @@ public ClusterState execute(ClusterState currentState) {
                 final String repoName = cloneEntry.repository();
                 final ShardGenerations shardGenerations = repoData.shardGenerations();
                 for (int i = 0; i < updatedEntries.size(); i++) {
-                    if (cloneEntry.snapshot().equals(updatedEntries.get(i).snapshot())) {
+                    final SnapshotsInProgress.Entry entry = updatedEntries.get(i);
+                    if (cloneEntry.repository().equals(entry.repository()) == false) {
+                        // different repo => just continue without modification
+                        continue;
+                    }
+                    if (cloneEntry.snapshot().getSnapshotId().equals(entry.snapshot().getSnapshotId())) {
                         final ImmutableOpenMap.Builder<RepositoryShardId, ShardSnapshotStatus> clonesBuilder =
                                 ImmutableOpenMap.builder();
-                        final InFlightShardSnapshotStates inFlightShardStates =
-                            InFlightShardSnapshotStates.forRepo(repoName, snapshotsInProgress.entries());
+                        final boolean readyToExecute = currentState.custom(
```

Comment:
`readyToExecute` kind of lack of meaning, maybe `startedSnapshotDeletion` (+ `anyMatch`)?

Reply:
++ this is 100% the better name, I'll leave it as is for a sec though so it's less confusing to review when drying things up (we use the same mechanics and var name in the normal snapshot shard assignments path ... and it's equally weird there in hindsight :))